packages/netsuite-adapter/src/change_validators/currency_exchange_rate.ts

```diff
@@ -0,0 +1,44 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { ChangeError, ChangeValidator, getChangeData, InstanceElement, isAdditionChange, isInstanceChange } from '@salto-io/adapter-api'
+import { collections, values } from '@salto-io/lowerdash'
+
+const { awu } = collections.asynciterable
+const { isDefined } = values
+
+const changeValidator: ChangeValidator = async changes => (
+  awu(changes)
+    .filter(isInstanceChange)
+    .filter(isAdditionChange)
+    .filter(async change => getChangeData<InstanceElement>(change).elemID.typeName === 'currency')
+    .map(async change => {
+      const instance = getChangeData(change)
```

Comment:
With the below suggestion you call `getChangeData` only once :)
```suggestion
    .map(async change => getChangeData<InstanceElement>(change))
    .filter(async instance => instance.elemID.typeName === 'currency')
    .map(async instance => {
```